tests/gitlab.py

```diff
@@ -31,6 +33,39 @@
 SAVE_GRAPHQL_RESULTS = False
 
 
+@dataclass
+class TestResult:
+    failed: int = 0
+    passed: int = 0
+    error: int = 0
+
+    @classmethod
+    def from_line(cls, line: str) -> TestResult:
+        self = TestResult()
+        for key in self.__annotations__:
+            match = re.search(rf"(\d+) {key}", line)
+            if match:
+                setattr(self, key, int(match.group(1)))
+        return self
+
+    @classmethod
+    def from_job_id(cls, job_id: str) -> TestResult:
+        report_link = cls._link_to_report(job_id)
+        raw_content = requests.get(report_link).text
+        result_pattern = r"= .* passed.*s \(\d.*\) ="
+        result_line = re.search(result_pattern, raw_content)
+        if not result_line:
+            print("No results yet.")
+            return TestResult()
+        return cls.from_line(result_line.group(0))
+
+    @staticmethod
+    def _link_to_report(job_id: str) -> str:
+        return (
+            f"https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/{job_id}/raw"
```

Comment:
nit: just make it a constant and run `LINK_TO_REPORT.format(job_id)`

Reply:
Right, changed in the force-push